### PR TITLE
Check for nil before accessing the stampedRef

### PR DESCRIPTION
**Pull request by emmjohnson**
[PR requirements]: https://github.com/vmware-tanzu/cartographer/blob/main/CONTRIBUTING.md#commit-message-and-pr-guidelines

## Changes proposed by this PR

closes # 

## Release Note

[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## PR Checklist

Note: Please do not remove items. Mark items as done `[x]` or use ~strikethrough~ if you believe they are not relevant

- [ ] Linked to a relevant issue. Eg: `Fixes #123` or `Updates #123`
- [ ] Removed non-atomic or `wip` commits
- [ ] Filled in the [Release Note](#Release-Note) section above 
- [ ] Modified the docs to match changes

### CODE REVIEW

File: pkg/controller/workload/reconciler.go [hunk 1]
```diff
@@ -313,7 +313,7 @@ func (r *Reconciler) cleanupOrphanedObjects(ctx context.Context, previousResourc
 		}
 		orphaned := true
 		for _, realizedResource := range realizedResources {
-			if realizedResource.StampedRef.GroupVersionKind() == prevResource.StampedRef.GroupVersionKind() &&
+			if realizedResource.StampedRef != nil && realizedResource.StampedRef.GroupVersionKind() == prevResource.StampedRef.GroupVersionKind() &&
```
Comment (cirocosta) [inline pkg/controller/workload/reconciler.go]: wdyt of instead breaking out of the loop preemptively?

ala

```
for _, realizedResource := range realizedResources {
  if realizedResource.StampedRef == nil {
     continue
  }
}
```

**Review by cirocosta — APPROVED** [review]
lgtm! just left a little nit

**Comment by cirocosta** [comment]
adding some context: this PR came from seeing that the container that runs the controller got a restart, not really sure what a good reproducible looks like though

the panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x140833b]

goroutine 569 [running]:
k8s.io/api/core/v1.(*ObjectReference).GroupVersionKind(...)
        k8s.io/api@v0.23.4/core/v1/objectreference.go:30
github.com/vmware-tanzu/cartographer/pkg/controller/workload.(*Reconciler).cleanupOrphanedObjects(0xc0004ab8c0, {0x1938118, 0xc0000301b0}, {0xc000d761a0, 0x5, 0x171a606}, {0xc000218000, 0x6, 0x7})
        github.com/vmware-tanzu/cartographer/pkg/controller/workload/reconciler.go:316 +0x3db
github.com/vmware-tanzu/cartographer/pkg/controller/workload.(*Reconciler).Reconcile(0xc0004ab8c0, {0x1938118, 0xc000ad5da0}, {{{0xc000d0acd0, 0x7}, {0xc000d0acc6, 0xa}}})
        github.com/vmware-tanzu/cartographer/pkg/controller/workload/reconciler.go:161 +0x2577
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0004c4b00, {0x1938118, 0xc000c2df80}, {{{0xc000d0acd0, 0x165f880}, {0xc000d0acc6, 0x413a94}}})
```

File: pkg/controller/workload/reconciler.go (hunk 1 above)
Comment (emmjohnson) [reply pkg/controller/workload/reconciler.go]: Done. And ported to deliverable 😅

**Pull request merged**